### ISSUE TITLE
"meeting started" notifications

### ISSUE THREAD

**Issue by emdej** [IsmaelMartinez/teams-for-linux#699]
In Microsoft native client for Linux there was an option to show notification saying "a meeting has started". It was very convenient. Without it it's likely to miss meetings.

![image](https://user-images.githubusercontent.com/12697118/223692758-fc272890-c593-4cfb-9908-52419b5426fd.png)

**Comment by jijojosephk** [comment]
Hi @emdej it's already there. You don't see it as a setting.

![image](https://user-images.githubusercontent.com/11662148/224477357-efef5d8d-82a8-4757-bd04-4ae586165915.png)

**Comment by emdej** [comment]
Hi @jijojosephk , I've upgraded to 1.0.52 and tried with and without `--enableDesktopNotificationsHack`. Chat notifications are there and meeting started notifications are missing. How can I debug it?

**Comment by emdej** [comment]
It's a deb package downloaded from github. Debian 11.6. Kernel 5.19.0-0.deb11.2-amd64. My colleague using teams-for-linux also face this issue. I'll check tomorrow what distibution they use.

**Comment by emdej** [comment]
Colleague told me:
Ubuntu 20.04, kernel: 5.14.0-1058. teams-for-linux: 1.0.49. Installed from snap.

**Comment by mtczekajlo** [comment]
On `Fedora release 37 (Thirty Seven) x86_64` doesn't work either.
I'm using the RPM release `Source: teams-for-linux-1.0.53-1.src.rpm`.

**Comment by emdej** [comment]
Without any change from user point of view, it has started to display meeting notifications today for me and my colleague. Did MS change something?

**Comment by jijojosephk** [comment]
@emdej I'm not sure. But I've been seeing this without any issues for quite sometime. Do you hear a weird audio notification as well ? :smile:

**Comment by emdej** [comment]
Very strange - it was working for just one day. And yes: I head weird audio notification too. I was hard to miss :). Currently it doesn't work again. Maybe there are multiple APIs for that and one works while the other one doesn't? Or there is some toggle for admin and he plays with it :D ?

**Comment by jijojosephk** [comment]
Can you clear storage once try?
`teams-for-linux --clearStorage` if it's not flatpak. Otherwise, `flatpak run com.github.IsmaelMartinez.teams_for_linux --clearStorage`. This should give you a fresh start.

**Comment by emdej** [comment]
Unfortunately it doesn't change anything. Can it be some server setting that it is enabled or disabled for particular user/domain?

**Comment by jijojosephk** [comment]
> Unfortunately it doesn't change anything. Can it be some server setting that it is enabled or disabled for particular user/domain?

I don't think so. Have  you by any chance enabled this flag? `--disableMeetingNotifications`. Are you using configuration file or command-line arguments? Or following what's installed and setup by default?

**Comment by emdej** [comment]
I don't use any command line arguments nor config file. Can I check, e.g. with wireshark if the notification arrives from server?

**Comment by jijojosephk** [comment]
@emdej can you confirm the version you have?

**Comment by emdej** [comment]
Still the same.

**Comment by emdej** [comment]
I tried the following approach: collected traffic with wireshark, then decrypted tls and hoped to see something meaningful. Unfortunately the data seems to be in some binary format and there is communication all the time, so it's hard to filter just the packages related to notification(s). Maybe you have some tips? Can some sort of event/traffic debugging be enabled from electron?

**Comment by IsmaelMartinez** [comment]
ok, something random but can you try:
* setting up a meeting using the "calendar" in ms Teams
* setting up a meeting via outlook calendar
* making it a "teams meeting"
* not making it a "teams meeting"
![image](https://user-images.githubusercontent.com/430992/234004419-5fd5fdf6-7880-4234-885f-394d37da9e4f.png)

The logic I use is that I rarely get meeting notifications when I set it up via the ms teams calendar when using the official MacOS app, so it might be a MS oddity.

**Comment by emdej** [comment]
I tried every option and it doesn't work in any of them.

I've installed official Android application and notifications do appear there every time. I don't know what API it uses...

**Comment by jijojosephk** [comment]
Do you hear a audio note?

**Comment by emdej** [comment]
No.

**Comment by emdej** [comment]
During the 2 days a few weeks ago everything was working: strange audio and popup in the corner. Before and after nothing works.

**Comment by jijojosephk** [comment]
That's weird!

**Comment by jijojosephk** [comment]
> I am using 1.0.88 on Ubuntu 22.04, with Unity interface (I am new to teams-for-linux, used official one before). I don't get notification and also no "calling sound". Everything else seems to be working, just this things not.. Made me miss (or be late) for few meetings...

Which package your'e using?

**Comment by andyrozman** [comment]
@jijojosephk 
Snap one, latest/stable from ismaelmartinez, revision 373

teams-for-linux            1.0.88                      373    latest/stable    ismaelmartinez  -

**Comment by jijojosephk** [comment]
Can you try anything other than `snap` please? `snap` drives us crazy.

**Comment by andyrozman** [comment]
Ok, will do... I usually install snap of any app, because it updates itself so I am always current, but I will install some other version and remove snap for now.

**Comment by andyrozman** [comment]
I installed with apt, I have version 1.0.90, but notification of incomming call still doesn't work:
On official client this would happen:
- special notification would be displayed, where you could accept or reject the call (it was usually at bottom of screen and it was little bit bigger than normal "text" notifications, which are usually displayed on top
- incomming call sound would be heard for configured amount of time

On Teams for linux (personal/direct call):
- works ok (same as official channel)

On Teams for linux (scheduled conference):
- no special notification
- just a short ping, which sometimes makes me miss a call/conference
- another note: on official client, when scheduled conference takes place, it calls you, same way as if someone would be trying to make direct call

**Comment by emdej** [comment]
> just a short ping, which sometimes makes me miss a call/conference

What do you mean by "ping"?

I've found that once someone joins a meeting there appears a difference in calendar view - the meeting gets dark.

**Comment by andyrozman** [comment]
So when someone calls you you get that melody and it runs for configured amount of time. When meeting start, Teams plays 1 short note (for about 1s) and thats it. If I am not behing computer (or if I am listening to music) this short note just passes me by... I am not sure if this is default behaviour of newer version of teams (have to test it on my Mac, but this will be only possible when I am back home), but this is not good enough, at least for me... If I don't get prolonged notification (like melody) I likely miss the metting.

**Comment by IsmaelMartinez** [comment]
can you check if is any of this two "wav" files? https://github.com/IsmaelMartinez/teams-for-linux/tree/develop/app/assets/sounds that "ping" you mean>?

**Comment by IsmaelMartinez** [comment]
@emdej and @andyrozman , did you try with the `watch` command that @storm9c1 send? @storm9c1, thanks a lot for the extensive investigation and help!

**Comment by IsmaelMartinez** [comment]
We can defo add a longer notification (on the back of a feature flag/config option) and see if that sorts the problem.

**Comment by andyrozman** [comment]
What about adding buttons to notification (and having notiifaction there for longer time) so that you can join call at once (without having to go to your calendar and join from there)?

**Comment by IsmaelMartinez** [comment]
that will require us handling the notifications. This is not something we do now but might be something we can try to add again in the future (2.0)

**Comment by emdej** [comment]
> @emdej and @andyrozman , did you try with the watch command that @storm9c1 send? @storm9c1, thanks a lot for the extensive investigation and help!

Yes, I tried it but as expected it's not the problem. State is all the time IDLE.

I didn't expect it to be the reason because other notifications arrive with no issue.

**Comment by jijojosephk** [comment]
@emdej, are you looking for the notification at the bottom right like in the official app? If so that's not possible.

**Comment by emdej** [comment]
> @emdej, are you looking for the notification at the bottom right like in the official app? If so that's not possible.

@IsmaelMartinez, Sorry for late response. I do not care if it is bottom or top right or left. Existing notifications related to e.g. chat messages are fine. "Join" button is nice to have, but not necessary - I can find the meeting in calendar but I need to know it's time to join. And this is indeed inspired by the unsupported Microsoft application. Why do you say it is not possible?

As #725 is a superset (superfeature?) of this feature (adds the persistency and "join" button), how that can be possible if this one is not?

**Comment by IsmaelMartinez** [comment]
Hi @emdej , what we mean is that we aren't in control of the notifications. The other issue/feature request, is to implement the code/logic to bring the notifications to our control. Easier said than done
